### PR TITLE
CI: add test_all_deps on docker build

### PR DESCRIPTION
**Pull request by junnplus**
## What type of PR is this? 

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] New Query Runner (Data Source) 
- [ ] New Alert Destination
- [ ] Other
- [x] CI

## How is this tested?

- [ ] Unit tests (pytest, jest)
- [ ] E2E Tests (Cypress)
- [ ] Manually
- [ ] N/A

### CODE REVIEW

**Comment by justinclift** [comment]
I don't think this will do what we're after.

pip seems to check things a bit differently using `pip install -r requirements_all_ds.txt`, compared to `pip3 install -r requirements.txt -r requirements_dev.txt -r requirements_all_ds.txt`.

Having all three of the requirements*.txt files on the same line seems to make a (positive) difference, letting it give a more thorough answer.

So even if we remove the option here, then we're not the thorough test of all dependencies.

**Comment by justinclift** [comment]
We could probably add an extra test to the Dockerfile, to add a specific test though:

```
RUN if [ "x$test_all_deps" != "x" ] ; then pip3 install -r requirements.txt -r requirements_dev.txt -r requirements_all_ds.txt ; fi
```

We'd need to add a matching `test_all_deps` ARG near the top of the file, and make sure we include this argument when running the test too.

**Comment by justinclift** [comment]
Ugh.  I keep on forgetting to add those comments using GitHub's "Review" process. :wink:

**Review by justinclift — CHANGES_REQUESTED** [review]
Pretty sure we'll need an `ARG` line added to the Dockerfile as well, so this new `test_all_deps` variable can be used.  Something like the lines of these ones:

* https://github.com/getredash/redash/blob/79b01406fc30925661a887d04f4176c56d491f6b/Dockerfile#L32-L35

Reference info, in case it's useful: :smile:

* https://docs.docker.com/engine/reference/builder/#arg

**Comment by junnplus** [comment]
> Pretty sure we'll need an ARG line added to the Dockerfile as well, so this new test_all_deps variable can be used. Something the lines of these ones:

yeah, I missed it.

File: Dockerfile [hunk 1]
```diff
@@ -86,8 +88,7 @@ WORKDIR /app
 ENV PIP_DISABLE_PIP_VERSION_CHECK=1
 ENV PIP_NO_CACHE_DIR=1
 
-# rollback pip version to avoid legacy resolver problem
-RUN pip install pip==20.2.4;
+RUN pip install pip==23.1.2;
```
Comment (justinclift) [inline Dockerfile:91]: Is the updated pip version here important?

Comment (junnplus) [reply Dockerfile:91]: Older versions of pip will silently suppress conflict messages.

see https://github.com/getredash/redash/actions/runs/5491313994/jobs/10007730439?pr=5865#step:3:1473

Comment (justinclift) [reply Dockerfile:91]: Interesting.  So that's what the comment above is about. :smile:

**Review by justinclift — APPROVED** [review]
Looks good to me. :smile:

**Pull request merged**